mobile/lib/shared/views/splash_screen.dart

```diff
@@ -20,21 +20,26 @@ class SplashScreenPage extends HookConsumerWidget {
 
     void performLoggingIn() async {
       bool isSuccess = false;
+      bool deviceIsOffline = false;
       if (accessToken != null && serverUrl != null) {
         try {
           // Resolve API server endpoint from user provided serverUrl
           await apiService.resolveAndSetEndpoint(serverUrl);
         } catch (e) {
           // okay, try to continue anyway if offline
+          deviceIsOffline = true;
```

Comment:
Consider adding a fallback or error message for the user in the case that the device is offline and login is unsuccessful. This would improve user experience by providing clear feedback. [medium]

Reply:
If the device is offline and the login is unsuccessful (i.e. there is no user information stored locally) then it will try to do an online login. If that also fails then it will redirect user back to the login page